edge-agent-sdk/src/commonMain/kotlin/org/hyperledger/identus/walletsdk/pollux/PolluxImpl.kt

```diff
@@ -493,6 +512,175 @@ class PolluxImpl(
+            throw PolluxError.UnsupportedTypeError(kty)
+        }
+
+        val publicKey = apollo.createPublicKey(
+            mapOf(
+                TypeKey().property to KeyTypes.EC,
+                CurveKey().property to crv,
+                CurvePointXKey().property to x,
+                CurvePointYKey().property to y
+            )
+        )
+        if (!publicKey.canVerify()) {
+            throw PolluxError.VerifyProofError()
+        }
+
+        if (!proof.containsKey("jws")) {
+            throw PolluxError.RevocationRegistryJsonMissingFieldError("Proof->jws")
+        }
+        val jwsArray = proof["jws"]?.jsonPrimitive?.content?.split(".") ?: throw PolluxError.InvalidJWTString()
+        if (jwsArray.size != 3) {
+            throw PolluxError.InvalidJWTString()
+        }
+
+        val payload = JsonObject(jsonObject.filterKeys { it != "proof" })
+        val encoded = encode(Json.encodeToString(payload))
+        val signaturePayload = "${jwsArray[0]}.$encoded"
+        val signature = jwsArray[2].base64UrlDecodedBytes
+
+        val verified = (publicKey as VerifiableKey).verify(
+            signaturePayload.toByteArray(),
+            signature
+        )
+        if (!verified) {
+            throw PolluxError.VerifyProofError()
+        }
+        return true
+    }
+
+    private fun encode(data: String): String {
+        val inputStream = ByteArrayInputStream(data.toByteArray())
+        val document = JsonDocument.of(inputStream)
+        val rdfDataset = JsonLd.toRdf(document).get()
+        val normalized = RdfNormalize.normalize(rdfDataset)
+        val writer = StringWriter()
+        val rdfWriter = Rdf.createWriter(com.apicatalog.jsonld.http.media.MediaType.N_QUADS, writer)
+        rdfWriter.write(normalized)
+
+        return writer.toString()
+    }
+
+    fun checkEncodedListRevoked(revocationRegistryJson: String, statusListIndex: Int): Boolean {
+        val jsonObject = (Json.parseToJsonElement(revocationRegistryJson) as JsonObject)
+        if (jsonObject.containsKey("credentialSubject")) {
+            val credentialSubject = jsonObject["credentialSubject"]!!.jsonObject
+            if (credentialSubject.containsKey("encodedList")) {
+                val encodedList = credentialSubject["encodedList"]?.jsonPrimitive?.content
+                if (encodedList != null) {
+                    val decodedBytes = Base64.getUrlDecoder().decode(encodedList)
+
+                    val byteArrayInputStream = ByteArrayInputStream(decodedBytes)
+                    val gzipInputStream = GZIPInputStream(byteArrayInputStream)
+                    val decompressedBytes = gzipInputStream.readBytes()
+                    if (statusListIndex > decompressedBytes.size) {
+                        throw PolluxError.StatusListOutOfBoundIndex()
+                    }
+                    return decompressedBytes[statusListIndex].toInt() == 1
```

Comment:
This is wrong. You need to convert this `decompressedBytes` to a list of Bits (Booleans) and then check the index.

Reply:
I updated this using the same approach as TS.